Source/WebGPU/WebGPU/TestPlumbing.swift

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+public typealias CppRefCounted = WebGPUTest.CppRefCounted
+
+public class TestPlumbing {
+    public static func getRefCount(obj: CppRefCounted) -> UInt32 {
+        // Just an API call to ensure swift does not optimize this out.
+        let refCount = obj.refCountProxy()
+        let _ = "The ref count is \(refCount)"
+        return refCount
+    }
+
+    public static func getRefCountOptimizedOut(obj: CppRefCounted) -> UInt32 {
+        obj.refCountProxy()
+    }
```

Comment:
This seems kinda brittle. I don't think we need it?

The goal is to determine that the object is allocated with the right refcount, right? If so, we want to export a Swift function that will allocate and return the object, and then we can test the refcount of the result directly in our caller TestWebKitAPI code.

Reply:
hmm.. okie.. I can remove that. I just wanted to highlight the optimization via a test but it's not important.